pkg/webhook/resources/persistentvolumeclaim/validator.go

```diff
@@ -66,6 +69,22 @@ func (v *pvcValidator) Delete(request *types.Request, oldObj runtime.Object) err
 		oldPVC.Status.Phase != "Terminating" {
 		if _, find := oldPVC.Annotations[util.AnnotationGoldenImage]; find {
 			if oldPVC.Annotations[util.AnnotationGoldenImage] == "true" {
+				// extra chance, if the corresponding vm image is deleted, we can delete the golden image PVC directly
+				imageName := oldPVC.Name
+				imageNS := oldPVC.Namespace
+				if v, find := oldPVC.Annotations[cdicommon.AnnPopulatorKind]; find && v == cdiv1.VolumeCloneSourceRef {
+					if image, find := oldPVC.Annotations[cdicommon.AnnEventSource]; find {
+						imageRaw := strings.Split(image, "/")
+						if len(imageRaw) == 2 {
+							imageNS = imageRaw[0]
+							imageName = imageRaw[1]
+						}
+					}
+				}
+				// don't care other errors, just check if the image is still there
+				if _, err := v.imageCache.Get(imageNS, imageName); apierrors.IsNotFound(err) {
+					return nil
+				}
```

Comment:
So this is saying, a golden image can be deleted if it's either:

* lost
* terminating or
* not associated to any vm images

correct?

Shouldn't this new check also be done outside of the `if oldPVC.Status.Phase != corev1.ClaimLost && oldPVC.Status.Phase != "Terminating"` conditions?

Reply:
Thanks for the suggestion!

I rewrite the goldenImageCheck mechanism.
Please help to recheck